main/fastcgi.c

```diff
@@ -688,7 +688,11 @@ int fcgi_listen(const char *path, int backlog)
 		if (!*host || !strncmp(host, "*", sizeof("*")-1)) {
 			sa.sa_inet.sin_addr.s_addr = htonl(INADDR_ANY);
 		} else {
+#ifdef HAVE_INET_PTON
+			inet_pton(AF_INET, host, &sa.sa_inet.sin_addr);
+#else
 			sa.sa_inet.sin_addr.s_addr = inet_addr(host);
+#endif
 			if (sa.sa_inet.sin_addr.s_addr == INADDR_NONE) {
```

Comment:
You also need to adjust this error check to check the return value of inet_pton instead.

Reply:
done